batchflow/models/tf/deeplab.py

```diff
@@ -0,0 +1,198 @@
+"""
+Liang-Chieh Chen, Yukun Zhu, George Papandreou, Florian Schroff, Hartwig Adam.
+"`Encoder-Decoder with Atrous Separable
+Convolution for Semantic Image Segmentation
+<https://arxiv.org/abs/1802.02611>`_"
+"""
+import tensorflow as tf
+
+from .layers import aspp
+from . import EncoderDecoder, Xception
+
+
+
+class DeepLab(EncoderDecoder):
+    """ DeepLab model archtecture.
+    Input is encoded through Xception backbone (entry, middle and exit flows),
+    then worked via Atrous Spatial Pyramid Pooling layer, and, finally, restored to original
+    spatial size with simple yet effective decoder module.
+
+    Parameters
+    ----------
+    inputs : dict
+        Dictionary with 'images' (see :meth:`~.TFModel._make_inputs`)
+
+    body : dict
+        entry : dict
```

Comment:
In general, `deeplab` has nothing to do with entry, middle, exit architecture. These are all Xception block parameters, but deeplab can be built upon ResNet backbone.

Reply:
Fairly speaking, current implementation of `DeepLab` class has very little to do with actual DeepLab architecture: this class is merely `EncoderDecoder` with multiple `embedding` blocks, default config values and some custom names. My plan is to:
1. Add `make_encoder` method to `Xception` class. That would allow to replicate one of the architectures from the article via `EncoderDecoder` only with config options: `Xception` as encoder and `ASPP` module as embedding.
2. Rename `entry`/`middle`/`exit` blocks in `DeepLab`, so it stays highly customizable, but without references to `Xception`

Reply:
So I decided to make `EncoderDecoder` class capable of handling multiple `embedding`'s. That definetely belongs to that class, not to `DeepLab`. It allowed to make `DeepLab` class by only changing `default_config`.
Commit is on the way..

Reply:
Deeplab config should not have entry, middle and exit options as these belong to Xception backbone, but not Deeplab itself. Deeplab can be built upon ResNet backbone (or any other architecture)